x/mongo/driver/topology/topology_test.go

```diff
@@ -160,7 +164,8 @@ func TestServerSelection(t *testing.T) {
 			t.Errorf("Timed out while trying to retrieve selected servers")
 		}
 
-		if err != context.Canceled {
+		want := ServerSelectionError{Wrapped: context.Canceled, Desc: topo.Description()}
+		if err.Error() != want.Error() {
```

Comment:
nit: This conditional can be changed to `assert.Equal(t, want, err, ...)`. Our `assert` library handles error comparisons by calling `Error()`: https://github.com/mongodb/mongo-go-driver/blob/a6cd2226c7836c1264a00b7101f751e95eeef570/internal/testutil/assert/assert.go#L19.

Reply:
done